File: app/java/src/main/java/com/example/production/domain/model/location/Location.java
package com.example.production.domain.model.location;

import lombok.*;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Location {
    private String locationCode;
    private String locationName;
    private LocationType locationType;
    private String parentLocationCode;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: app/csharp/src/ProductionManagement.Domain/Models/Location/Location.cs
namespace ProductionManagement.Domain.Models.Location;

/// <summary>
/// 場所マスタ
/// </summary>
public class Location
{
    private string _locationTypeValue = "倉庫";

    public required string LocationCode { get; set; }

    public required string LocationName { get; set; }

    /// <summary>
    /// 場所区分（データベース格納用の日本語文字列）
    /// </summary>
    public string LocationTypeValue
    {
        get => _locationTypeValue;
        set => _locationTypeValue = value;
    }

    /// <summary>
    /// 場所区分（ドメインロジック用の Enum）
    /// </summary>
    public LocationType LocationType
    {
        get => LocationTypeExtensions.FromDisplayName(_locationTypeValue);
        set => _locationTypeValue = value.GetDisplayName();
    }

    public string? ParentLocationCode { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }
}
